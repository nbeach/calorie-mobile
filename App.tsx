import React, {Component} from 'react';
import {Platform, StyleSheet, Text, View} from 'react-native';
import {Home} from "./src/component/presentation/Home";


interface Props {}
export default class App extends Component<Props> {
  render() {
    return (
      <View style={styles.container}>
        <Home></Home>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 40,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});


