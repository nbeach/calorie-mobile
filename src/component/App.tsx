import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Provider} from "react-redux"
import {initializeStore} from "../store/store";
import {HomeContainer} from "./container/HomeContainer";

export const App = () =>
    <Provider store={initializeStore()}>
      <View style={styles.container}>
        <HomeContainer></HomeContainer>
      </View>
    </Provider>


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'stretch',
  },
});


