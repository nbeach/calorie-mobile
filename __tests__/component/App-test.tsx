import React from 'react';
import renderer from 'react-test-renderer';
import {App} from "../../src/component/App";

describe(App.name, () => {

    it("displays the app", () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

});

