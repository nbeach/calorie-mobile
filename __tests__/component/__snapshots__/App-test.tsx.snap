// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App displays the app 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#e4ebee",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        placeholder="Item"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#AAAAAA",
            "borderStyle": "solid",
            "fontSize": 18,
            "width": "60%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        placeholder="Cals"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#AAAAAA",
            "borderStyle": "solid",
            "fontSize": 18,
            "width": "15%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1b5fff",
            "borderColor": "#EEEEEE",
            "borderStyle": "solid",
            "borderWidth": 1,
            "fontSize": 18,
            "justifyContent": "center",
            "opacity": 1,
            "width": "15%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;
